Core/Core/MobileCore.cs

```diff
@@ -213,9 +212,29 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
             throw new ServiceModuleInstanceNotFoundException(String.Format("No instance has been registered for interface {0}", serviceClass.Name));
         }
 
-        public ServiceConfiguration GetServiceConfiguration(String type)
+        public ServiceConfiguration[] GetServiceConfigurationByType(String type)
+        {
+            List<ServiceConfiguration> listOfConfigs = new List<ServiceConfiguration>();
+            foreach (var item in servicesConfig)
+            {
+                if (item.Key == type)
+                {
+                    listOfConfigs.Add(item.Value);
+                }
+            }
+            ServiceConfiguration[] arrayOfConfigs = listOfConfigs.ToArray();
+            return arrayOfConfigs;
+        }
+
+        public ServiceConfiguration GetFirstServiceConfigurationByType(String type)
         {
             return servicesConfig.ContainsKey(type) ? servicesConfig[type] : null;
```

Comment:
I don't think the implementation here is right? This should call `GetServiceConfigurationByType` and return the first one in the array?